apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/PgNumericTests.cs

```diff
@@ -0,0 +1,445 @@
+                "0.00123"
+            )] string text,
+           bool negate)
+        {
+            if (negate)
+            {
+                text = "-" + text;
+            }
+
+            PgNumeric numeric = PgNumeric.Parse(text);
+            Assert.Equal(text, numeric.ToString());
+
+            // Check that TryParse works too
+            Assert.True(PgNumeric.TryParse(text, out var numeric2));
+            Assert.Equal(numeric, numeric2);
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidStringInputData))]
+        public void Parse_Invalid(string input)
+        {
+            Assert.Throws<FormatException>(() => PgNumeric.Parse(input));
+            bool success = PgNumeric.TryParse(input, out PgNumeric number);
+            Assert.Equal(PgNumeric.Zero, number);
+            Assert.False(success);
+        }
+
+        [Theory]
+        [InlineData("NaN", "NaN")]
+        [InlineData("1.00", "1.00")]
+        [InlineData("1.000000", "1.000000")]
+        [InlineData("99999.999999000", "99999.999999000")]
+        [InlineData("-0.000001", "-0.000001")]
+        [InlineData("100", "100")]
+        [InlineData("1.33333333333333333", "1.33333333333333333")]
+        public void ToString_(string input, string expected)
+        {
+            PgNumeric pgNumeric = PgNumeric.Parse(input);
+            Assert.Equal(expected, pgNumeric.ToString());
+        }
+
+        [Theory]
+        [MemberData(nameof(FromDecimalData))]
+        public void FromDecimal(decimal input, string expectedValue)
+        {
+            var numeric = PgNumeric.FromDecimal(input);
+            Assert.Equal(expectedValue, numeric.ToString());
+        }
+
+        [Theory]
+        [InlineData("0.00")]
+        [InlineData("1.00")]
+        [InlineData("1.01")]
+        [InlineData("12345.123456000")]
+        [InlineData("-0.000001")]
+        [InlineData("1234567891.123456789123")]
+        [InlineData("-12345678912.12345678912")]
+        [InlineData("123456789123456789.12345678912")]
+        [InlineData("12345678912345678912345678.1234567891234567891234")]
+        [InlineData("9999999999999999999999999999.999999999999999999999999")]
+        [InlineData("100")]
+        [InlineData("1.33333333333333333")]
+        public void ToDecimal(string input)
+        {
+            var numeric = PgNumeric.Parse(input);
+            // decimal.Parse will silently lose precision if input has more significant digits than decimal can handle.
+            decimal expected = decimal.Parse(input, CultureInfo.InvariantCulture);
+            decimal actual = numeric.ToDecimal(LossOfPrecisionHandling.Truncate);
+            Assert.Equal(expected, actual);
+            // Conversion via the explicit conversion should do the same thing.
+            decimal actual2 = (decimal)numeric;
+            Assert.Equal(expected, actual2);
+        }
+
+        [Theory]
+        [InlineData("123456789012345678900.123456789", "123456789012345678900.12345679")]
```

Comment:
Hmm... this example still doesn't look like truncation. It should be converting to "123456789012345678900.12345678". Let me know if you'd like me to pull the code and debug this.